Added exploit for CVE-2023-25194 (#18134)
rapid7/metasploit-framework: documentation/modules/exploit/multi/http/apache_druid_cve_2023_25194.md

## Vulnerable Application

Apache Kafka is an open-source distributed event streaming platform that is used for real-time data streaming and processing.
Kafka clients are a set of Java libraries that allow you to produce and consume messages from Apache Kafka.

In the version prior to 3.3.2, there is a JNDI injection issue in the Apache Kafka clients if an attacker is able to set the

> `sasl.jaas.config` property for any of the connector's Kafka clients to `com.sun.security.auth.module.JndiLoginModule`.

Reviewer: Isn't this enabled by default?

Author: Since Apache Kafka 3.0.0, users are allowed to specify these properties in connector configurations for Kafka Connect clusters running with out-of-the-box configurations. Before Apache Kafka 3.0.0, users may not specify these properties unless the Kafka Connect cluster has been reconfigured with a connector client override policy that permits them.

From: https://lists.apache.org/thread/vy1c7fqcdqvq5grcqp6q5jyyb302khyz

Reviewer: Please add this to the documentation then :)

Author: This link is in  the references for the module. I tried to focus the document in how it affects the Druid.

It will allow the server to connect to the attacker's LDAP server and deserialize the LDAP response, which the attacker can
use to execute java deserialization gadget chains on the Kafka connect server. Attacker can cause unrestricted deserialization
of untrusted data (or) RCE vulnerability when there are gadgets in the classpath.

> As Apache Druid dependeds on kafka-clients to connect to one of its datasources, making it out of the box vulnerable if using any

Reviewer 2: ```suggestion
As Apache Druid depends on kafka-clients to connect to one of its data sources, making it out of the box vulnerable if using any
```

affected version.

The below docker compose file can be used to build a vulnerable container.

> ```doocker-compose

Reviewer 2: This seems unnecessary as docker-compose is not the language (syntax highlighting won't be applied).
```suggestion
```
```

version: '2'
services:
 web:
   image: vulhub/apache-druid:25.0.0
   ports:
    - "8888:8888"
```

## Verification Steps

Metasploit:

1. `./msfconsole`
1. `use exploit/multi/http/apache_druid_cve_2023_25194`

> 1. `set rhosts <rhost>`

Reviewer 2: The following were required to be changed from their default value in order to test the docker compose file target, I figure they should be included in the verification steps.
```suggestion
1. `set rhosts <rhost>`
1. `set target <target>`
1. `set srvhost <srvhost>`
1. `set SSL <true or false>`
```
